### ISSUE TITLE
Error dialogs should be auto capitalized

### ISSUE THREAD

**Issue by lostdusty** [fyne-io/fyne#4876]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

According to Go's wiki:
> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context.

However, our context (error dialogs) are usually showed to the end/final user, and I believe texts in error dialogs should start capitalized, and end with a period.

[See discussion on Discord.](https://discord.com/channels/953020122690359327/953031065331695666/1243630292372295681)

### Is it possible to construct a solution with the existing API?

Yes

### Describe the solution you'd like to see.

By default, every error dialog should start capitalized and end with a period. I believe most languages follows this "convention" (?).
To disable, maybe you could use a function like:
```go
a := dialog.NewError(fmt.Errorf("something went very wrong in our side, error code: %v", errInt))
a.DisableAutoCaps()
a.Show()
```

**Comment by Jacalz** [comment]
I am a bit torn on this one. If you need to show something more readable, would it not be better to use the information dialog? You cloud attach some extra information and be more helpful for the user.

**Comment by andydotxyz** [comment]
Given that this is just a helper for what may be unexpected errors it seems like something reasonable to add, given that Go insists we have no capital letters at the beginning of the text.

**Comment by lostdusty** [comment]
Maybe later I can submit a PR :)

**Comment by andydotxyz** [comment]
Fixed in v2.5.1
